haystack/dataclasses/breakpoints.py

```diff
@@ -6,91 +6,65 @@
 from typing import Optional, Union
 
 
-@dataclass
+@dataclass(frozen=True)
 class Breakpoint:
     """
     A dataclass to hold a breakpoint for a component.
+
+    :param component_name: The name of the component where the breakpoint is set.
+    :param visit_count: The number of times the component must be visited before the breakpoint is triggered.
+    :param debug_path: Optional path to store the state of the pipeline when the breakpoint is hit.
+        This is useful for debugging purposes, allowing you to inspect the state of the pipeline at the time of the
+        breakpoint and to resume execution from that point.
     """
 
     component_name: str
     visit_count: int = 0
+    debug_path: Optional[str] = None
 
-    def __hash__(self):
-        return hash((self.component_name, self.visit_count))
-
-    def __eq__(self, other):
-        if not isinstance(other, Breakpoint):
-            return False
-        return self.component_name == other.component_name and self.visit_count == other.visit_count
-
-    def __str__(self):
-        return f"Breakpoint(component_name={self.component_name}, visit_count={self.visit_count})"
 
-    def __repr__(self):
-        return self.__str__()
```

Comment:
From what I could tell these are the default `__str__` and `__repr__` methods that come with a dataclass.